python-sdk/docs/configurations.rst

```diff
@@ -23,15 +23,103 @@ or by updating Airflow's configuration
+
+.. code:: shell
+
+   AIRFLOW__ASTRO_SDK__DATAFRAME_ALLOW_UNSAFE_STORAGE = True
+
+or by updating Airflow's configuration
+
+.. code:: shell
+
+   [astro_sdk]
+   dataframe_allow_unsafe_storage = True
+
+Configuring the storage integration for Snowflake
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A storage integration is a Snowflake object that stores a generated identity and access management (IAM) entity for your external cloud storage, along with an optional set of allowed or blocked storage locations (Amazon S3, Google Cloud Storage, or Microsoft Azure). Cloud provider administrators in your organization grant permissions on the storage locations to the generated entity. This option allows users to avoid supplying credentials when creating stages or when loading or unloading data.
+
+Read more at: `Snowflake storage integrations <https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html>`_
+
+.. code:: shell
+
+   AIRFLOW__ASTRO_SDK__SNOWFLAKE_STORAGE_INTEGRATION_AMAZON = "aws_integration"
+   AIRFLOW__ASTRO_SDK__SNOWFLAKE_STORAGE_INTEGRATION_GOOGLE = "gcp_integration"
+
+or by updating Airflow's configuration
+
+.. code:: shell
+
+   [astro_sdk]
+   snowflake_storage_integration_amazon = "aws_integration"
+   snowflake_storage_integration_google = "gcp_integration"
+
+Configuring the table autodetect row count
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Following configuration indicates how many file rows should be loaded to infer the table columns types. This defaults to 1000 rows.
+
+.. code:: shell
+
+   AIRFLOW__ASTRO_SDK__LOAD_TABLE_AUTODETECT_ROWS_COUNT = 1000
+
+or by updating Airflow's configuration
+
+.. code:: shell
+
+   [astro_sdk]
+   load_table_autodetect_rows_count = 1000
+
+Configuring the RAW SQL maximum response size
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Reduce responses sizes returned by aql.run_raw_sql to avoid trashing the Airflow DB if the BaseXCom is used.
+
+.. code:: shell
+
+   AIRFLOW__ASTRO_SDK__RUN_RAW_SQL_RESPONSE_SIZE = 1
+
+or by updating Airflow's configuration
+
+.. code:: shell
+
+   [astro_sdk]
+   run_raw_sql_response_size = 1
+
+
+Configuring the Dataset inlets/outlets
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Astro SDK automatically adds inlets and outlets for all the operators if DATASET is supported (Airflow >=2.4).
+
+Users can over-ride it on a task level by adding inlets and outlets themselves. However, for users who want to use SDK with Airflow 2.4 and above but do not want to leverage Data-aware scheduling, auto-addition of inlets/outlets would be annoying so this new config setting will provide an escape hatch for such users.
```

Comment:
```suggestion
While users can override it on a task level by adding inlets and outlets, this might be inconvenient for some users who do not want to leverage Data-aware scheduling. Such users can set the following config to `False` to disable auto addition of inlets and outlets
```